rewrite trivial views

dannyroberts opened the pull request:
The following views just emit docs of a certain type by domain, which can be trivally replaced by a call to `domain/docs`. I'm going to rewrite them one by one, and then we can delete them in a future PR.
- [x] `cloudcare/application_access`
- [x] `facilities/registries_by_domain`
- [x] `fixtures/data_types_by_domain`
- [x] ~~`userreports/report_configs_by_domain`~~ this is in a different db, so domain/docs doesn't exist, and having an extra view here is way cheaper than in the main db
- [x] ~~`groupexport/by_domain`~~ there's a call that doesn't group by domain (just gets all of them in any domain), which you can't do with domain/docs, so this isn't worth doing
- [x] ~~`commtrack/domain_config`~~ there's a call that doesn't group by domain
- [x] `sms/forwarding_rule`
- [x] ~~`groups/by_domain`~~ this one uses cachehq; I'm don't feel like the way it's done right now is very good to begin with, but I'm going to just leave it be.
- [x] `reminders/survey_by_domain`

dannyroberts committed:
pull out cloudcare/application_access call to dbaccessor

add test

dimagimon commented on corehq/apps/cloudcare/tests/test_dbaccessors.py:
```diff
@@ -0,0 +1,27 @@
+            application_access_objects.append(o)
+            self.assertIn(
+                get_application_access_for_domain(domain).to_json(),
+                [o.to_json() for o in application_access_objects]
```
F812 list comprehension redefines 'o' from line 19

dannyroberts replied:
(i'm ok with this)

dannyroberts committed:
remove usage of cloudcare/application_access

dannyroberts committed:
pull out and test facilities/registries_by_domain call

had to fix method for adding new timestamps

apparently no one has saved a FacilityRegistry or Facility since before
April

dimagimon commented on corehq/apps/facilities/tests.py:
```diff
@@ -0,0 +1,28 @@
+            o.save()
+            objects.append(o)
+            self.assertItemsEqual(
+                [o.to_json() for o in get_facility_registries_in_domain(domain)],
```
F812 list comprehension redefines 'o' from line 13

dimagimon commented on corehq/apps/facilities/tests.py:
```diff
@@ -0,0 +1,28 @@
+            o.save()
+            objects.append(o)
+            self.assertItemsEqual(
+                [o.to_json() for o in get_facility_registries_in_domain(domain)],
```
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dimagimon replied:
F812 list comprehension redefines 'o' from line 20

dannyroberts committed:
remove usage of facilities/registries_by_domain

dannyroberts committed:
pull out and test calls to fixtures/data_types_by_domain

dannyroberts committed:
remove calls to fixtures/data_types_by_domain

dannyroberts committed:
pull out userreports/report_configs_by_domain calls

- add tests for these dbaccessors
- add test for untested dbaccessor

dannyroberts committed:
pull out calls to sms/forwarding_rule

and test
and clean up imports

dannyroberts committed:
remove calls to sms/forwarding_rule

dannyroberts committed:
pull out calls to reminders/survey_by_domain and test

dannyroberts committed:
remove calls to reminders/survey_by_domain

dannyroberts commented:
OK I think I'm done for now. Good to go (on my end) once tests pass. @snopoke @benrudolph @kaapstorm buddy

dannyroberts commented:
Frees up 5 views to be deleted at our convenience!

czue commented:
kicked build, but looked like it might be a real import thing

dannyroberts commented:
:white_check_mark: These are pretty rote, but worth reviewing commit-by-commit

dannyroberts commented:
@benrudolph @kaapstorm

kaapstorm commented:
:+1:

dannyroberts commented:
FYI i'm merging this into `fixtures-spike` and then reverting https://github.com/dimagi/commcare-hq/commit/0a8ab2108d25ef6dd4df4a89ae2cd00fc3f3791c on that branch. I'll have to do the same thing for the PR we do to remove the fixtures view. (Basically, I want this optimization for now in the short term, but when fixtures goes to its own db, the original code is better.)

benrudolph commented:
clever

benrudolph commented on corehq/apps/reminders/dbaccessors.py:
```diff
@@ -0,0 +1,9 @@
+def get_surveys_in_domain(domain):
+    from corehq.apps.reminders.models import Survey
+    return sorted(Survey.view(
```
shouldn't these already be sorted given that the domain is the same, survey should be in lexicographical order? though i suppose this might be emitting the id and not the name

benrudolph replied:
looking below, it doesn't seem that the get sorted, any reason for doing so? probably not a big deal

dannyroberts replied:
The old view sorted by survey name (by emitting [domain, name]); since the new view doesn't, we sort it in memory to achieve the same effect.

For more of a blow by blow see https://github.com/dimagi/commcare-hq/commit/7c870d3b9a8c44877a66d17e2c74cac4954332fe and https://github.com/dimagi/commcare-hq/commit/4efd6257e76288a55df95dba47c742126c8a2a0f

benrudolph commented:
looks good. minor comment that really is pretty inconsequential. merging

dannyroberts commented:
Thanks!
